src/Agent.Listener/Environment.cs

```
@@ -0,0 +1,213 @@
+
+        //An array of paths, which if found in the file system means the capability is present.
+        public string[] Paths { get; private set; }
+    }
+
+    struct ShellCapability
+    {
+        public ShellCapability(string name, string command, string commandArgs)
+        {
+            Name = name;
+            Command = command;
+            CommandArgs = commandArgs;
+        }
+
+        //Name of the capability as requried by the task definition.
+        public string Name { get; private set; }
+
+        //Name of an executable file, which can determine if a capability is present.
+        //The script is expected to print on the standard output the value for the capability.
+        public string Command { get; private set; }
+
+        //Arguments passed to Command.
+        public string CommandArgs { get; private set; }
+    }
+
+    //extensions which simplify adding constant capabilities into a list
+    static class ListExtensions
+    {
+        public static void Add(this IList<Capability> list, string name, string tool = null, string[] paths = null)
+        {            
+            list.Add(new Capability(name, tool, paths));
+        }
+
+        public static void Add(this IList<ShellCapability> list, string name, string command, string commandArgs = null)
+        {
+            list.Add(new ShellCapability(name, command, commandArgs));
+        }
+    }
+
+    public sealed class Environment : AgentService, IEnviroment
+    {
+        private static readonly List<Capability> _regularCapabilities =
+            new List<Capability>
+            {
+                {
+                    "AndroidSDK",
+                    "android",
+                    new string[]
+                    {
+                        System.Environment.GetEnvironmentVariable("ANDROID_STUDIO") + "/tools/android",
+                        System.Environment.GetEnvironmentVariable("HOME") + "/Library/Developer/Xamarin/android-sdk-macosx/tools/android"
```

Thinking more about it, I don't like to call functions in a type initializer. I'll have to move this code to run latter and catch all exceptions.